discovery_engine_core/core/src/document.rs

```diff
@@ -222,6 +223,27 @@ pub(crate) fn document_from_article(
     })
 }
 
+/// Removes documents where the title or url is duplicated.
+pub(crate) fn dedup_documents(documents: &mut Vec<Document>) {
+    // discard dups in the title keeping only the best ranked
+    documents.sort_unstable_by(
+        |doc1, doc2| match doc1.resource.title.cmp(&doc2.resource.title) {
+            Ordering::Equal => doc1.resource.rank.cmp(&doc2.resource.rank),
```

Comment:
there is indeed a `published_date` on articles but not sure whether that is a better way to choose between (say in this case) 2 articles with the same title. assuming we want to keep the "better" one (which may not necessarily be the earliest one), it seems to me sensible to let rank guide the choice?

Reply:
Yeah, I suppose this might also be a question we would want to direct at pink? 🤔 I'm certainly not equipped to answer it 😅 Maybe it also doesn't matter here, as we're talking about exact duplicates.

Reply:
At the moment we want to try with the rank but this will be probably a subject for many tests to see what works better.